README.md

```diff
@@ -1,117 +1,167 @@
-Puppet Strings comments, and then use Strings to generate a REFERENCE.md in your
-module. For details on how to add code comments and generate documentation with
-Strings, see the [Puppet Strings documentation][2] and [style guide][3].
+To test that everything was configured properly, we can run any Bolt plan that runs a PuppetDB query, for example:
 
-If you aren't ready to use Strings yet, manually create a REFERENCE.md in the
-root of your module directory and list out each of your module's classes,
-defined types, facts, functions, Puppet tasks, task plans, and resource types
-and providers, along with the parameters for each.
+```
+# /root/Projects/local_plan/plans/test.pp
+
+plan local_plan::test(
+){
+  $query_results = puppetdb_query("nodes[]{}")
+  out::message("Hello world from the Bolt Server, query results: ${query_results}")
+}
+```
 
-For each element (class, defined type, function, and so on), list:
+Run the Bolt plan:
 
-* The data type, if applicable.
-* A description of what the element does.
-* Valid values, if the data type doesn't make it obvious.
-* Default value, if any.
+`bolt plan run local_plan::test`
 
-For example:
+We should see the PuppetDB query results in the terminal, and if we inspect the `puppetdb-access.log` there should be a log with a call to the local PuppetDB with a 200 Ok HTTP status:
+
+```
+$ less /var/log/puppetlabs/puppetdb/puppetdb-access.log
 
+127.0.0.1 - - [01/Nov/2022:15:56:21 +0000] "POST /pdb/query/v4 HTTP/1.1" 200 1793 "-" "HTTPClient/1.0 (2.8.3, ruby 2.7.6 (2022-04-12))" 99 21 -
 ```
-### `pet::cat`
 
-#### Parameters
+### Running a Plan via taskplan from the Primary Server
+
+We recommend to install the [`taskplan` module](https://forge.puppet.com/modules/reidmv/taskplan).
+
+- The intended use for `taskplan` is to run Plans in the Bolt server, especifically by telling it to run it on a target node, in this case, our Bolt server.
+
+**Running a Plan on the Bolt Server**
+
+This is an overview of the internal process when you offload the Plan execution from the PE Primary server to the Bolt server:
 
-##### `meow`
+1. Someone requests Orchestrator to run a task on the Bolt server
+1. The task runs on the Bolt server
+1. The task starts Bolt with the "bolt plan run" command
+1. Bolt starts and runs the plan
 
-Enables vocalization in your cat. Valid options: 'string'.
+![bolt-server-process](diagrams/bolt-server-exec-processes.png "Bolt server execution process")
+
+**Example 1**
+
+In this example, we will use `bolt task run` to run the `taskplan` Task on the Bolt server.
+
+Required parameters:
+
+- Choose one of your existing basic Plans, or create one that receives a parameter, `message` in this case.
+- Use the certname of your Bolt Server
+
+From the PE Primary server CLI we will run:
 
-Default: 'medium-loud'.
+```
+puppet task run taskplan --params '{"plan":"a-test-plan", "params":{"message": "Hello world!"}, "debug":true}' -n insert-here-the-bolt-server-certname
```

Comment:
Does that command line parameter actually work? If I remember correctly, we didn't succeed making it work and resorted to putting the parameters in a json param file.

Reply:
This command line works, we used it in our load test script.

We used the params JSON file with the Curl approach

`curl -d "@test_param.json" --insecure --header "$auth_header" "$uri"`

Reply:
@mcka1n I thought we used the params file for the `puppet tsk run` approach too because we couldn't make the command line work?